pkg/asset/agent/image/agentimage.go

```diff
@@ -103,29 +108,75 @@ func (a *AgentImage) updateIgnitionImg(ignition []byte) error {
 		return err
 	}
 
-	ignitionImgPath := filepath.Join(a.tmpPath, "images", "ignition.img")
-	fi, err := os.Stat(ignitionImgPath)
+	// Defining a struct for the ignition info
+	type IgnInfo struct {
+		File   string `json:"file"`
+		Length int64  `json:"length"`
+		Offset int64  `json:"offset"`
+	}
+	var ignInfo IgnInfo
+
+	// Reading the ignition details from igninfo.json file into the struct
+	ignInfoJSONPath := filepath.Join(a.tmpPath, "coreos", "igninfo.json")
+	ignInfoJSONData, err := os.ReadFile(ignInfoJSONPath)
 	if err != nil {
+		logrus.Warnf("Failed to read json %s", ignInfoJSONPath)
+		return err
+	}
+	if err := json.Unmarshal(ignInfoJSONData, &ignInfo); err != nil {
+		logrus.Warnf("Failed to umarshal json: %s", err)
 		return err
 	}
 
-	// Verify that the current compressed ignition archive does not exceed the
-	// embed area (usually 256 Kb)
-	if len(ignitionBuff) > int(fi.Size()) {
-		return fmt.Errorf("ignition content length (%d) exceeds embed area size (%d)", len(ignitionBuff), fi.Size())
+	// Fetching the ignition details from the struct
+	ignImagePath := filepath.Join(a.tmpPath, ignInfo.File)
+	ignStartOffset := ignInfo.Offset
+	ignMaxLength := ignInfo.Length
+
+	// Verify that the compressed ignition buffer archive does not exceed the embed area (usually 256 Kb)
+	if ignMaxLength == 0 {
+		// Checking the file size if no ignition length is provided
+		fileInfo, err := os.Stat(ignImagePath)
+		if err != nil {
+			return err
+		}
+
+		if len(ignitionBuff) > int(fileInfo.Size()) {
+			return fmt.Errorf("ignition content length (%d) exceeds embed area size (%d)", len(ignitionBuff), fileInfo.Size())
+		}
+	} else if int64(len(ignitionBuff)) > ignMaxLength {
+		return fmt.Errorf("Ignition content length (%d) exceeds embed area size (%d)", len(ignitionBuff), ignMaxLength)
 	}
```

Comment:
The embedded ignition archive is always 0-padded (to 256 Kb) to allow coreos-installer to work properly, so I think it would be better to keep the original logic, ie: 

```
if len(ignitionBuff) > int(fi.Size()) {
...
```

Reply:
@andfasano , for cdboot.img files we are trying to compare the length of ignition archive only to the maximum length it should be and not with the entire file size because the cdboot.img is having other data as well apart from the ignition content and the file size is very large relatively with default use case